### PR TITLE
TMA-491: Ruby SDK, bricks should support to release/rollout restricte…

### PR DESCRIPTION
**Pull request by korczis**
…d flag

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Changelog
- [ ] Build passing

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/775/ : FAILURE in 28s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/731/ : FAILURE in 3m 26s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/501/ : FAILURE in 27s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/778/ : FAILURE in 14s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/734/ : FAILURE in 3m 51s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/504/ : FAILURE in 1m 22s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/782/ : FAILURE in 20s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/738/ : FAILURE in 3m 38s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/508/ : FAILURE in 39s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/784/ : FAILURE in 18s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/740/ : FAILURE in 4m 05s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/510/ : UNSTABLE in 13m 11s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/786/ : SUCCESS in 36s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/742/ : FAILURE in 3m 32s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/512/ : UNSTABLE in 14m 25s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/787/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/743/ : SUCCESS in 45m 42s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/513/ : UNSTABLE in 13m 26s

File: lib/gooddata/models/blueprint/to_wire.rb [hunk 1]
```diff
@@ -97,7 +97,10 @@ def self.dataset_to_wire(project, dataset)
             title: GoodData::Model.title(dataset),
             anchor: anchor_to_wire(project, dataset),
             attributes: attributes_to_wire(project, dataset),
-            facts: DatasetBlueprint.facts(dataset).map { |f| fact_to_wire(dataset, f) },
+            facts: DatasetBlueprint.facts(dataset).map do |f|
```
Comment (vhtien) [inline lib/gooddata/models/blueprint/to_wire.rb]: I don't know why we should change it. I think we should keep the diff as small as we can, right?

File: lib/gooddata/models/metadata.rb [hunk 2]
```diff
@@ -163,6 +163,36 @@ def deprecated=(flag)
       else
         fail 'You have to provide flag as either 1 or "1" or 0 or "0" or true/false'
       end
+
+      set_flag('deprecated', flag)
+    end
+
+    def production
+      if meta['isProduction'] == '1' || get_flag('production')
```
Comment (vhtien) [inline lib/gooddata/models/metadata.rb]: why don't we return at this line?

File: lib/gooddata/models/metadata.rb [hunk 3]
```diff
@@ -282,5 +312,20 @@ def unlisted=(flag)
     def validate
       true
     end
+
+    def get_flag(flag)
```
Comment (vhtien) [inline lib/gooddata/models/metadata.rb]: do you think `has_flag` is a better name?

File: lib/gooddata/models/metadata.rb [hunk 4]
```diff
@@ -282,5 +312,20 @@ def unlisted=(flag)
...
+      meta['flags'] && meta['flags'].include?(flag)
+    end
+
+    def set_flag(flag, value)
+      meta['flags'] = [] unless meta['flags']
+
+      if (value == '1' || value == 1 || value == true) && !meta['flags'].include?(flag)
```
Comment (vhtien) [inline lib/gooddata/models/metadata.rb]: could we replace `meta['flags'].include?(flag)` with `has_flag(flag)` for readability?

File: lib/gooddata/models/metadata.rb [hunk 5]
```diff
@@ -282,5 +312,20 @@ def unlisted=(flag)
...
+        meta['flags'].push(flag)
+        meta['flags'].sort!
+      elsif !value && meta['flags'].include?(flag)
```
Comment (vhtien) [inline lib/gooddata/models/metadata.rb]: could we replace `meta['flags'].include?(flag)` with `has_flag(flag)` for readability?

File: lib/gooddata/models/blueprint/to_wire.rb (hunk 1 above)
Comment (korczis) [inline lib/gooddata/models/blueprint/to_wire.rb]: Fixed.

File: lib/gooddata/models/metadata.rb (hunk 2 above)
Comment (korczis) [inline lib/gooddata/models/metadata.rb]: Fixed.

**Pull request merged**